src/crystal/event_loop/iocp.cr

```diff
@@ -228,6 +228,14 @@ class Crystal::EventLoop::IOCP < Crystal::EventLoop
   end
 
   def write(file_descriptor : Crystal::System::FileDescriptor, slice : Bytes) : Int32
+    if file_descriptor.system_append?
+      # we must explicitly seek to the end of the file; trying to set the
+      # overlapped structure offset to UInt64::MAX will correctly write at the
+      # end of file, but the following seeks will report an invalid position (it
+      # only counts the written bytes)
+      file_descriptor.system_seek(0, IO::Seek::End)
```

Comment:
We still need to set the offset of the overlapped struct, because this seek isn't atomic with the following write. Another thread writing can mess with the file pointer.

Maybe we should seek to the end after the write, too 🤔

Reply:
Maybe seeking to the end when we open the file would be enough? That would spare a synchronous seek on every write, but append is only supposed to seek to the end of the file along with write (atomically) :face_exhaling: